pact/coin-contract/coin-faucet.pact

```diff
@@ -0,0 +1,94 @@
+(module coin-faucet FAUCET-GOVERNANCE
+
+  "'coin-faucet' represents Kadena's Coin Faucet Contract."
+
+  (defcap FAUCET-GOVERNANCE () true)
+
+  (use coin)
+
+  ; --------------------------------------------------------------------------
+  ; Schemas and Tables
+  ; --------------------------------------------------------------------------
+
+  (defschema history
+    @doc "Table to record of the behavior of addresses. Last transaction time,   \
+    \ total coins earned, and total coins returned are recorded per transaction. "
+    last-tx-time:time
+    total-coins-earned:decimal
+    total-coins-returned:decimal
+    )
+
+  (deftable history-table: {history})
+
+  ; --------------------------------------------------------------------------
+  ; Capabilities
+  ; --------------------------------------------------------------------------
+
+  (defcap HISTORY ()
+    "Enforces faucet-guard when looking up history"
+    (enforce-guard (faucet-guard)))
+
+  ; --------------------------------------------------------------------------
+  ; Constants
+  ; --------------------------------------------------------------------------
+
+  (defconst FAUCET_ACCOUNT 'faucet-account)
+  (defconst COIN_PER_REQUEST 1.0)
+  (defconst WAIT_TIME_PER_REQUEST 3600.0)
+  (defconst EPOCH  (time "1970-01-01T00:00:00Z"))
+
+  ; --------------------------------------------------------------------------
+  ; Coin Faucet Contract
+  ; --------------------------------------------------------------------------
+
+  (defun faucet-guard:guard () (create-module-guard 'faucet-admin))
```

Comment:
It really makes more sense for this to be a `defconst` but I know that didn't work. Need to revisit in pact